# Title

*Description*

Jupyter Notebook written by Ben Fisher on 5 December 2024 <br>
**benjamin.s.fisher@usace.army.mil**

### Imports

For libraries not installed use: !pip install \<library\> to install from Notebook

In [7]:
import os, warnings, datetime, time

import pandas as pd
import numpy as np
import regex as re

In [8]:
import bs4 as bs
import lxml

### Testing

The following links (UNC and URL) direct to the same ProjectWise Directory

In [45]:
target_unc = 'pw://EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/P{34f8c63d-73bf-4469-a87e-56f725808937}/'
target_url = 'pw://EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/Military/02_IWA/FY26&space;MCN&space;H2406&space;Repair&space;Whole&space;House&space;Building&space;1255,&space;Iwakuni/08_Design&space;Documents/02_Draft&space;DBRFP/Working/'

##### Testing with OS

In [46]:
try:
    dir_list = os.listdir(target_unc)
except OSError as e:
    print(e)

[WinError 123] The filename, directory name, or volume label syntax is incorrect: 'pw://EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/P{34f8c63d-73bf-4469-a87e-56f725808937}/'


WinError 123 - seems to be raised because of the colons in the unc (OSError)

In [47]:
try:
    dir_list = os.listdir(target_url)
except FileNotFoundError as e:
    print(e)

[WinError 3] The system cannot find the path specified: 'pw://EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/Military/02_IWA/FY26&space;MCN&space;H2406&space;Repair&space;Whole&space;House&space;Building&space;1255,&space;Iwakuni/08_Design&space;Documents/02_Draft&space;DBRFP/Working/'


WinError 3 - appears to be raised because the dir can't be found?

###### Testing with pathlib

In [54]:
import pathlib

In [60]:
pathlib.Path(target_unc)

WindowsPath('pw:/EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/P{34f8c63d-73bf-4469-a87e-56f725808937}')

In [61]:
pathlib.Path(target_url)

WindowsPath('pw:/EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01/Documents/Military/02_IWA/FY26&space;MCN&space;H2406&space;Repair&space;Whole&space;House&space;Building&space;1255,&space;Iwakuni/08_Design&space;Documents/02_Draft&space;DBRFP/Working')

In [67]:
try:
    list(pathlib.Path(target_unc).iterdir())
except OSError as e:
    print(e)

[WinError 123] The filename, directory name, or volume label syntax is incorrect: 'pw:\\EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01\\Documents\\P{34f8c63d-73bf-4469-a87e-56f725808937}'


In [68]:
try:
    list(pathlib.Path(target_url).iterdir())
except FileNotFoundError as e:
    print(e)

[WinError 3] The system cannot find the path specified: 'pw:\\EPOJW2SN02AOA01.poj.ds.usace.army.mil:PWPOJ01\\Documents\\Military\\02_IWA\\FY26&space;MCN&space;H2406&space;Repair&space;Whole&space;House&space;Building&space;1255,&space;Iwakuni\\08_Design&space;Documents\\02_Draft&space;DBRFP\\Working'
